src/application/controllers/settings/update-settings.ts

```diff
@@ -0,0 +1,43 @@
+import { badRequest, notAcceptable, serverError, success, type HttpRequest, type HttpResponse } from '@/application/helpers'
+import { type Validation, type Controller } from '@/application/protocols'
+import { type UpdateSettings } from '@/domain/use-cases'
+
+export class UpdateSettingsController implements Controller {
+  private readonly updateSettings: UpdateSettings
+  private readonly validation: Validation
+
+  constructor ({ updateSettings, validation }: UpdateSettingsController.Dependencies) {
+    this.updateSettings = updateSettings
+    this.validation = validation
+  }
+
+  async handle (httpRequest: HttpRequest): Promise<HttpResponse> {
+    try {
+      const error = this.validation.validate({ ...httpRequest.body })
```

Comment:
o httpRequest.body já é um objeto não precisa desestruturar pode passar diretamente.
```suggestion
      const error = this.validation.validate(httpRequest.body)
```